src/test/lib/logical_query_plan/stored_table_node_test.cpp

```diff
@@ -438,16 +436,121 @@ TEST_F(StoredTableNodeTest, HasMatchingUniqueColumnCombination) {
   EXPECT_EQ(_stored_table_node->unique_column_combinations().size(), 1);
 
   // Negative test.
+  // Columns are empty.
+  EXPECT_THROW(_stored_table_node->has_matching_ucc({}), std::logic_error);
+
+  // There is no matching UCC.
   EXPECT_FALSE(_stored_table_node->has_matching_ucc({_b}));
   EXPECT_FALSE(_stored_table_node->has_matching_ucc({_c}));
   EXPECT_FALSE(_stored_table_node->has_matching_ucc({_b, _c}));
 
+  if constexpr (HYRISE_DEBUG) {
+    // Columns are not part of output_expressions() (i.e., pruned).
+    _stored_table_node->set_pruned_column_ids({ColumnID{0}});
+    EXPECT_THROW(_stored_table_node->has_matching_ucc({_a}), std::logic_error);
+  }
+
   // Test exact match.
+  _stored_table_node->set_pruned_column_ids({});
   EXPECT_TRUE(_stored_table_node->has_matching_ucc({_a}));
 
   // Test superset of column ids.
   EXPECT_TRUE(_stored_table_node->has_matching_ucc({_a, _b}));
   EXPECT_TRUE(_stored_table_node->has_matching_ucc({_a, _c}));
 }
 
+TEST_F(StoredTableNodeTest, OrderDependenciesSimple) {
+  EXPECT_TRUE(_stored_table_node->order_dependencies().empty());
+
+  // Create ODs from order constraints.
+  _table_a->add_soft_order_constraint({{ColumnID{0}}, {ColumnID{1}}});
+  _table_a->add_soft_order_constraint({{ColumnID{0}}, {ColumnID{2}}});
+  EXPECT_EQ(_table_a->soft_order_constraints().size(), 2);
+
+  const auto& order_dependencies = _stored_table_node->order_dependencies();
+  EXPECT_EQ(order_dependencies.size(), 2);
+  EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_a}, {_b}}));
+  EXPECT_TRUE(order_dependencies.contains(OrderDependency{{_a}, {_c}}));
+}
+
+TEST_F(StoredTableNodeTest, OrderDependenciesPrunedColumns) {
+  // Discard ODs that involve pruned columns.
+  _table_a->add_soft_order_constraint({{ColumnID{0}}, {ColumnID{1}}});
+  _table_a->add_soft_order_constraint({{ColumnID{0}}, {ColumnID{2}}});
+  EXPECT_EQ(_table_a->soft_order_constraints().size(), 2);
+
+  _stored_table_node->set_pruned_column_ids({ColumnID{0}, ColumnID{2}});
+  const auto& order_dependencies = _stored_table_node->order_dependencies();
```

Comment:
(If I remember correctly, we get a vector her)

Not that it matters but we are copying the vector first and use a reference to the temporarily materialized object. Why not `EXPECT_TRUE(_stored_table_node->order_dependencies().empty());`